### PR TITLE
Base structure for field dependencies

### LINKED ISSUE #106: Master details(field dependencies)

### PR DESCRIPTION
**Pull request by IgorZiegler**
- Form check all items if item has dependencies form resolve them
 - Add example in radiogroup

fixes #106

### CODE REVIEW

File: frontend/blocks/base/base.js [hunk 1]
```diff
@@ -76,6 +76,46 @@ export default class Base {
     this.popover('show');
   }
 
+  on() {
+    $(this).on(...arguments);
+
+    return this;
+  }
```
Comment (102) [inline frontend/blocks/base/base.js]: Can't we use something like this instead?
```js
on(..args) {
  $(this.on(..args));

  return this;
}
```

**Comment by 102** [comment]
```
$ ./node_modules/.bin/eslint frontend/*

~/Development/smartforms/frontend/blocks/form/form.js
  46:31  error  A space is required after '{'   object-curly-spacing
  46:58  error  A space is required before '}'  object-curly-spacing

✖ 2 problems (2 errors, 0 warnings)
```

File: frontend/blocks/form/form.js [hunk 2]
```diff
@@ -9,8 +11,47 @@ export default class Form extends Container {
...
+
+    block[isResolved ? 'show' : 'hide']();
+  }
+
+  get value() {
+    const result = {};
+
+    this.items.filter(block => !block.isHidden).forEach(block => Object.assign(result, block.value));
+
+    return this.config.name ? {[this.config.name]: result} : result;
```
Comment (102) [inline frontend/blocks/form/form.js]:
```
$ ./node_modules/.bin/eslint frontend/*

~/Development/smartforms/frontend/blocks/form/form.js
  46:31  error  A space is required after '{'   object-curly-spacing
  46:58  error  A space is required before '}'  object-curly-spacing

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by 102** [comment]
@IgorZiegler you've also forgot to add issue number (guess it's #106) in the PR comment

File: frontend/blocks/form/form.js [hunk 3]
```diff
@@ -9,8 +10,42 @@ export default class Form extends Container {
...
+    return this.config.name ? { [this.config.name]: result } : result;
+  }
+
+  set value(val) {
+    // TODO realization of set value function
+  }
+
   validate() {
-    return this.items.some(block => !block.validate());
+    return this.items.filter(block => !block.isHidden).every(block => block.validate());
```
Comment (mvahrameev) [inline frontend/blocks/form/form.js]: ```.every(block => block.isHidden || block.validate())```

File: frontend/blocks/base/base.js [hunk 4]
```diff
@@ -76,6 +76,46 @@ export default class Base {
...
   }
 
+  on(...args) {
+    $(this).on(...args);
+
+    return this;
+  }
+
+  trigger(...args) {
+    $(this).trigger(...args);
```
Comment (102) [inline frontend/blocks/base/base.js]: Why do you want to use `$(this)` instead of `this.el`?

File: frontend/blocks/base/base.js [hunk 5]
```diff
@@ -76,6 +76,46 @@ export default class Base {
...
+    this.isHidden = false;
+    this.el.show();
+    this.trigger('show', this);
+  }
+
+  get isHidden() {
+    return this._isHidden;
+  }
+
+  set isHidden(val) {
```
Comment (mvahrameev) [inline frontend/blocks/base/base.js]: why?

**Pull request merged**